src/wp-includes/class-wp-scripts.php

```diff
@@ -907,9 +907,19 @@ private function is_delayed_strategy( $strategy ) {
 	 * @return string The best eligible loading strategy.
 	 */
 	private function get_eligible_loading_strategy( $handle ) {
-		$eligible = $this->filter_eligible_strategies( $handle );
+		$intended = (string) $this->get_data( $handle, 'strategy' );
 
-		// Bail early once we know the eligible strategy is blocking.
+		// Bail early if there is no intended strategy.
+		if ( ! $intended ) {
+			return '';
+		}
+
+		// If the intended strategy is 'defer', limit the initial list of eligibles.
+		$initial = ( 'defer' === $intended ) ? array( 'defer' ) : null;
```

Comment:
Why is this the case for `defer` and not the case for `async`? It's not clear to me what this is for.

Reply:
I think in the case of `async`, `defer` is still an eligible (fallback) strategy. In the other direction that doesn't apply, so if a script is intended to have `defer`, it would never be changed ("fall back") to `async`.

Reply:
Ah, because if `$intended` is `async` then _both_ `defer` and `async` would be possible strategies. The `async` strategy can fallback to `defer`.

Reply:
(@felixarntz I replied to myself before I saw your reply 😄)